src/main/java/net/mcreator/workspace/references/ReferencesFinder.java

```diff
@@ -0,0 +1,273 @@
+				elements.add(me);
+			} else if (anyValueMatches(ge, String.class, e -> e.isAnnotationPresent(BlocklyXML.class),
+					(a, t) -> t.contains(localizationKey))) {
+				elements.add(me);
+			}
+		}
+
+		return elements;
+	}
+
+	/**
+	 * Checks if values acquired from any valid fields or methods meet the specified condition.
+	 *
+	 * @param source    The object to extract values from.
+	 * @param clazz     The class of values to be checked.
+	 * @param validIf   The predicate used to check if a field/method is considered valid.
+	 * @param condition The predicate defining the condition that the acquired values should pass.
+	 * @param <T>       The type of values to be checked.
+	 * @return Whether any value extracted from valid fields/methods on the {@code source} object pass the condition.
+	 */
+	public static <T> boolean anyValueMatches(@Nullable Object source, Class<T> clazz,
+			Predicate<AccessibleObject> validIf, BiPredicate<AccessibleObject, T> condition) {
+		if (source == null)
+			return false;
+
+		for (Field field : source.getClass().getFields()) {
+			if (!Modifier.isStatic(field.getModifiers()) && validIf.test(field)) {
+				try {
+					field.setAccessible(true);
+					if (checkValue(field.get(source), field, clazz, validIf, condition))
+						return true;
+				} catch (IllegalAccessException | IllegalArgumentException ignored) {
+				}
+			}
+		}
+		for (Method method : source.getClass().getMethods()) {
+			if (!Modifier.isStatic(method.getModifiers()) && validIf.test(method)) {
+				try {
+					method.setAccessible(true);
+					if (checkValue(method.invoke(source), method, clazz, validIf, condition))
+						return true;
+				} catch (IllegalArgumentException | ReflectiveOperationException ignored) {
+				}
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Checks if the value (or its components) acquired from the given field/method passes the specified condition.
+	 *
+	 * @param value     The extracted value that should be checked.
+	 * @param field     The field/method the {@code value} was acquired.
+	 * @param clazz     The class of values to be checked.
+	 * @param validIf   The predicate used to check if the field/method is considered valid.
+	 * @param condition The predicate defining the condition that the acquired values should pass.
+	 * @param <T>       The type of values to be checked.
+	 * @return Whether the provided value or any value extracted from valid fields/methods on the {@code value} object
+	 * passes the provided condition.
+	 */
+	@SuppressWarnings("unchecked")
+	private static <T> boolean checkValue(@Nullable Object value, AccessibleObject field, Class<T> clazz,
+			Predicate<AccessibleObject> validIf, BiPredicate<AccessibleObject, T> condition) {
+		if (value == null)
+			return false;
+
+		if (clazz.isInstance(value)) { // value of specified type
+			return condition == null || condition.test(field, (T) value);
+		} else if (value instanceof Collection<?> list) { // list of values

```

Comment:
Here you should check more universal object Iterable, also handle arrays.

Reference: #4025

Reply:
As noted in referenced PR, checking array with instanceof will not work so change in https://github.com/MCreator/MCreator/pull/3500/commits/1e9bc92e142138dbafbd384fc1de9bcbedf36f5e will not do the trick